gax/src/main/java/com/google/api/gax/rpc/testing/FakeApiExceptionFactory.java

```diff
@@ -0,0 +1,131 @@
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.rpc.testing;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.rpc.AlreadyExistsException;
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.CancelledException;
+import com.google.api.gax.rpc.DeadlineExceededException;
+import com.google.api.gax.rpc.FailedPreconditionException;
+import com.google.api.gax.rpc.UnavailableException;
+import com.google.api.gax.rpc.UnknownException;
+
+/**
+ * A factory class that returns the corresponding type of exception class from the given status
+ * code.
+ *
+ * <p>For more information about the status codes returned by the underlying grpc exception see
+ * https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/Status.java
+ */
+@BetaApi
+public class FakeApiExceptionFactory {
+  @BetaApi
+  public static ApiException createException(
+      Throwable cause, FakeStatusCode.Code statusCode, boolean retryable) {
+    FakeStatusCode grpcStatusCode = FakeStatusCode.of(statusCode);
+    switch (statusCode) {
+      case CANCELLED:
+        return new CancelledException(cause, grpcStatusCode, retryable);
+        //      case NOT_FOUND:
+        //        return new NotFoundException(cause, grpcStatusCode, retryable);
+      case UNKNOWN:
+        return new UnknownException(cause, grpcStatusCode, retryable);
+        //      case INVALID_ARGUMENT:
+        //        return new InvalidArgumentException(cause, grpcStatusCode, retryable);
+      case DEADLINE_EXCEEDED:
+        return new DeadlineExceededException(cause, grpcStatusCode, retryable);
+      case ALREADY_EXISTS:
+        return new AlreadyExistsException(cause, grpcStatusCode, retryable);
+        //      case PERMISSION_DENIED:
```

Comment:
What do all this commented string mean? I don't see any related FIXME or TODO.

Reply:
It's because I copied the full set from the real exception factory and commented out the ones that I didn't define in the fake enum.